hw/top_earlgrey/doc/design/top_generation.md

```diff
@@ -0,0 +1,83 @@
+# Earl Grey Top Generation
+
+Earl Grey is assembled/generated by our in-house integration tool [topgen](../../../../util/topgen/README.md).
+
+## Autogenerated Files warning
+
+The topgen tool autogenerates a number of modules and other files when it runs.
+A number of these files have been committed into the repository for browsing
+purposes. Please do not modify these files manually:
+- `rtl/top_earlgrey.sv`
+- The crossbar modules
+- The interrupt controller
+
+## How a top module is created
+
+The top module `rtl/top_earlgrey.sv` is created by `topgen` using the following command:
+
+```bash
+../../util/topgen.py -t data/top_earlgrey.hjson -o . -v
+```
+
+The data declared in `data/top-earlgrey.hjson` is the single-source-of-truth
+for the autogeneration of collateral for the Earl Grey design.
+
+Among others, the topgen command generates these files:
+
+- `rtl/top_earlgrey.sv`: Top module generated from the template
+    `data/top_earlgrey.sv.tpl` with the configuration file
+    `data/top_earlgrey.hjson`
+- `rtl/xbar_main.sv` and `rtl/tl_main_pkg.sv`: Crossbar module. Earl Grey has
+    only one main crossbar. (tlgen library is used to generate these files.)
+- `rtl/rv_plic*.sv` and `data/rv_plic.hjson`: Interrupt controller module.
```

Comment:
This item is incorrect: rv_plic is generated by ipgen: there is no reg_rv_plic.py tool anymore.